src/Forms/TreeDropdownField.php

```diff
@@ -453,6 +453,14 @@ public function tree(HTTPRequest $request)
         /** @var DataObject|Hierarchy $obj */
         $obj = null;
         $sourceObject = $this->getSourceObject();
+
+        // Precache numChildren count if possible.
+        if ($this->getNumChildrenMethod() == 'numChildren') {
+            // We're not calling `Hierarchy::prepopulateTreeDataCache()` because we're not customising results based
+            // on version or Fluent locales. So there would be no performance gain from additional caching.
+            Hierarchy::prepopulate_numchildren_cache($sourceObject);
```

Comment:
Seems fine... It'd be nice having `prepopulatedTreeDataCache` only because that provides extension points for performance enhancements in the future but I understand that would probably make it slower in this case 😧 .